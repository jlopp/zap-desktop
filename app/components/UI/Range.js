import React from 'react'
import { asField } from 'informed'
import styled, { withTheme } from 'styled-components'

const Input = styled.input`
  overflow: hidden;
  width: 100%;
  appearance: none;
  outline: none;

  ::-webkit-slider-runnable-track {
    height: 8px;
    appearance: none;
    background-color: ${props => props.theme.colors.gray};
    margin-top: -1px;
  }

  ::-webkit-slider-thumb {
    width: 8px;
    appearance: none;
    height: 8px;
    cursor: ew-resize;
    background: ${props => props.theme.colors.white};
    box-shadow: -1000px 0 0 1000px orange;
  }
`

const Range = asField(({ fieldState, fieldApi, ...props }) => {
  const { value } = fieldState
  const { setValue, setTouched } = fieldApi
  const { onChange, onBlur, initialValue, forwardedRef, ...rest } = props
  return (
    <Input
      {...rest}
      type="range"
      min={0}
      max={100}
      step={1}
      ref={forwardedRef}
      value={value || initialValue || '0'}
      onChange={e => {
        setValue(e.target.value)
        if (onChange) {
          onChange(e)
        }
      }}
      onBlur={e => {
        setTouched()
        if (onBlur) {
          onBlur(e)
        }
      }}
    />
  )
})

export default withTheme(Range)
