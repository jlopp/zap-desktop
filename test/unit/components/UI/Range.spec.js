import React from 'react'
import { Form } from 'informed'
import Range from 'components/UI/Range'
import renderer from 'react-test-renderer'
import { dark } from 'themes'
import { ThemeProvider } from 'styled-components'

describe('component.UI.Range', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <ThemeProvider theme={dark}>
          <Form>
            <Range field="name" />
          </Form>
        </ThemeProvider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
