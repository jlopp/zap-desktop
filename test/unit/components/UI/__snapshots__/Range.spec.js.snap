// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.Range should render correctly 1`] = `
.c0 {
  overflow: hidden;
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
}

.c0::-webkit-slider-runnable-track {
  height: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: #959595;
  margin-top: -1px;
}

.c0::-webkit-slider-thumb {
  width: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  height: 8px;
  cursor: ew-resize;
  background: #ffffff;
  box-shadow: -1000px 0 0 1000px orange;
}

<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <input
    className="c0"
    max={100}
    min={0}
    onBlur={[Function]}
    onChange={[Function]}
    step={1}
    type="range"
    value="0"
  />
</form>
`;
